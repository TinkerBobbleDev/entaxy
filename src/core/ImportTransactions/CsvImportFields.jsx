import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'

const styles = () => ({
  root: {
    display: 'flex',
    'flex-direction': 'column',
    margin: '15px'
  },
  textField: {
    margin: '0px 20px 15px 0px'
  }
})

const CsvImportFields = ({
  classes,
  institution,
  values,
  handleChange
}) => {
  switch (institution) {
    case 'BMO':
      return (
        <div className={classes.root}>
          <Typography variant="body1" gutterBottom={true}>
            Fill out the following field
          </Typography>
          <TextField
            label="Currency"
            inputProps={{
              'aria-label': 'Currency',
              required: true,
              maxLength: 6
            }}
            value={values.ticker}
            name="ticker"
            onChange={handleChange}
            className={classes.textField}
          />
        </div>
      )
    case 'TD':
    case 'Tangerine':
      return (
        <div className={classes.root}>
          <Typography variant="body1" gutterBottom={true}>
            A few things are missing from the CSV file
          </Typography>
          <TextField
            label="Account"
            inputProps={{
              'aria-label': 'Account',
              required: true,
              maxLength: 100
            }}
            value={values.account}
            name="account"
            helperText="The name of the account (e.g. Checking, Savings, RRSP, etc)"
            onChange={handleChange}
            className={classes.textField}
          />
          <TextField
            label="Currency"
            inputProps={{
              'aria-label': 'Currency',
              required: true,
              maxLength: 6
            }}
            value={values.ticker}
            name="ticker"
            onChange={handleChange}
            className={classes.textField}
          />
        </div>
      )
    default:
      return null
  }
}

CsvImportFields.propTypes = {
  institution: PropTypes.string.isRequired
}

export default withStyles(styles)(CsvImportFields)
