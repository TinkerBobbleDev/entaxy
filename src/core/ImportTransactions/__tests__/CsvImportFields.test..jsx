import React from 'react'
import renderer from 'react-test-renderer'
import CsvImportFields from '../CsvImportFields'

describe('CsvImportFields', () => {
  const mockHandleChange = jest.fn()
  const props = {
    classes: { }
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot when institution is not specified', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        classes={{ ...props }}
      />
    ))
    expect(component.toJSON()).toBeNull()
  })

  it('matches snapshot for BMO', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        institution="BMO"
        values={{ ticker: 'CAD' }}
        classes={{ ...props }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for TD', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        institution="TD"
        values={{ ticker: 'CAD' }}
        classes={{ ...props }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for Tangerine', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        institution="Tangerine"
        values={{ ticker: 'CAD' }}
        classes={{ ...props }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
