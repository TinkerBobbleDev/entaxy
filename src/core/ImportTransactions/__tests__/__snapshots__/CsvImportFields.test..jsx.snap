// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsvImportFields matches snapshot for BMO 1`] = `
<div
  className="CsvImportFields-root-1"
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-gutterBottom-30"
  >
    Fill out the following field
  </p>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-49 MuiFormLabel-filled-53 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46 MuiInputLabel-shrink-45"
      data-shrink={true}
    >
      Currency
    </label>
    <div
      className="MuiInputBase-root-69 MuiInput-root-56 MuiInput-underline-60 MuiInputBase-formControl-70 MuiInput-formControl-57"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Currency"
        className="MuiInputBase-input-79 MuiInput-input-64"
        disabled={false}
        maxLength={6}
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="CAD"
      />
    </div>
  </div>
</div>
`;

exports[`CsvImportFields matches snapshot for TD 1`] = `
<div
  className="CsvImportFields-root-1"
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-gutterBottom-30"
  >
    A few things are missing from the CSV file
  </p>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-49 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46"
      data-shrink={false}
    >
      Account
    </label>
    <div
      className="MuiInputBase-root-69 MuiInput-root-56 MuiInput-underline-60 MuiInputBase-formControl-70 MuiInput-formControl-57"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Account"
        className="MuiInputBase-input-79 MuiInput-input-64"
        disabled={false}
        maxLength={100}
        name="account"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root-86"
    >
      The name of the account (e.g. Checking, Savings, RRSP, etc)
    </p>
  </div>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-49 MuiFormLabel-filled-53 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46 MuiInputLabel-shrink-45"
      data-shrink={true}
    >
      Currency
    </label>
    <div
      className="MuiInputBase-root-69 MuiInput-root-56 MuiInput-underline-60 MuiInputBase-formControl-70 MuiInput-formControl-57"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Currency"
        className="MuiInputBase-input-79 MuiInput-input-64"
        disabled={false}
        maxLength={6}
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="CAD"
      />
    </div>
  </div>
</div>
`;

exports[`CsvImportFields matches snapshot for Tangerine 1`] = `
<div
  className="CsvImportFields-root-1"
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-gutterBottom-30"
  >
    A few things are missing from the CSV file
  </p>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-49 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46"
      data-shrink={false}
    >
      Account
    </label>
    <div
      className="MuiInputBase-root-69 MuiInput-root-56 MuiInput-underline-60 MuiInputBase-formControl-70 MuiInput-formControl-57"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Account"
        className="MuiInputBase-input-79 MuiInput-input-64"
        disabled={false}
        maxLength={100}
        name="account"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root-86"
    >
      The name of the account (e.g. Checking, Savings, RRSP, etc)
    </p>
  </div>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-49 MuiFormLabel-filled-53 MuiInputLabel-root-42 MuiInputLabel-formControl-43 MuiInputLabel-animated-46 MuiInputLabel-shrink-45"
      data-shrink={true}
    >
      Currency
    </label>
    <div
      className="MuiInputBase-root-69 MuiInput-root-56 MuiInput-underline-60 MuiInputBase-formControl-70 MuiInput-formControl-57"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Currency"
        className="MuiInputBase-input-79 MuiInput-input-64"
        disabled={false}
        maxLength={6}
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="CAD"
      />
    </div>
  </div>
</div>
`;
